frontend/src/main/scala/bloop/bsp/BloopBspServices.scala

```diff
@@ -421,6 +424,48 @@ final class BloopBspServices(
     Task.eval { originToCompileStores.remove(params.originId); () }.executeAsync
   }
 
+  def linkProjects(
+      userProjects: ProjectMapping,
+      state: State,
+      compileArgs: List[String],
+      originId: Option[String],
+      logger: BspServerLogger
+  ): BspResult[bsp.CompileResult] = {
+    import bloop.engine.tasks.LinkTask.{linkJS, linkMainWithNative}
+    userProjects._2.platform match {
+      case platform @ Js(config, _, _) =>
+        val cmd = Commands.Link(List(userProjects._2.name))
+        val targetDir = ScalaJsToolchain.linkTargetFrom(userProjects._2, config)
+        linkJS(cmd, userProjects._2, state, None, targetDir, platform).map { state =>
+          val result = if (state.status == ExitStatus.LinkingError) {
+            Right(bsp.CompileResult(originId, bsp.StatusCode.Error, None, None))
+          } else {
+            Right(bsp.CompileResult(originId, bsp.StatusCode.Ok, None, None))
+          }
+          (state, result)
+        }
+      case Jvm(_, _, _, _, _, _) =>
+        compileProjects(userProjects :: Nil, state, compileArgs, originId, logger)
```

Comment:
Alternatively here we can do the same as in the interpreter when it hits a JVM project on link. I'm not sure if there is much difference.